### PR TITLE
refactor(exthost) - V2 - SCM - Remove unused SCMRawResource properties

### PR DESCRIPTION
**Pull request by bryphe**
The protocol for the `SCMRawResource` object has changed. In the previous extension host, the `SCMRawResource` type looked like this:
```
export type SCMRawResource = [
	number /*handle*/,
	UriComponents /*resourceUri*/,
	string[] /*icons: light, dark*/,
	string /*tooltip*/,
	boolean /*strike through*/,
	boolean /*faded*/,

	string | undefined /*source*/,
	string | undefined /*letter*/,
	ThemeColor | null /*color*/
];
```
https://github.com/onivim/vscode-exthost/blob/8ebec0403c19a038932368df8c89e4676e923dd2/src/vs/workbench/api/node/extHost.protocol.ts#L621

In the new extension host, the `SCMRawResource` type is now:
```
export type SCMRawResource = [
	number /*handle*/,
	UriComponents /*resourceUri*/,
	UriComponents[] /*icons: light, dark*/,
	string /*tooltip*/,
	boolean /*strike through*/,
	boolean /*faded*/
];
```
https://github.com/onivim/vscode-exthost/blob/50bef147f7bbd250015361a4e3cad3305f65bc27/src/vs/workbench/api/common/extHost.protocol.ts#L794

Luckily - it doesn't seem like we are actually these extra properties anywhere (as far as I can tell - @glennsl , you might want to check me on that!)

Otherwise - the SCM API looks pretty much the same, so it should be straightforward to bring to the v2 exthost, aside from this one protocol change.

### CODE REVIEW

File: src/Extensions/ExtHostClient_SCM.re [hunk 1]
```diff
@@ -65,9 +62,7 @@ module Decode = {
         `String(tooltip),
         `Bool(strikeThrough),
         `Bool(faded),
-        source,
-        letter,
-        color,
+        ..._additionalArgs,
```
Comment (glennsl) [inline src/Extensions/ExtHostClient_SCM.re:65]: Just in case?

**Comment by glennsl** [comment]
> it doesn't seem like we are actually these extra properties anywhere 

No, I don't think they were yet. And if they were I can;t see how this would compile. I wonder where the letter and color comes from now though.

File: src/Extensions/ExtHostClient_SCM.re (hunk 1 above)
Comment (bryphe) [reply src/Extensions/ExtHostClient_SCM.re:65]: If I changed it to this: 
```
`List([
        `Int(handle),
        uri,
        `List(icons),
        `String(tooltip),
        `Bool(strikeThrough),
        `Bool(faded),
      ]
```
it wouldn't pattern match the 'v1' protocol.

Comment (bryphe) [reply src/Extensions/ExtHostClient_SCM.re:65]: I'll remove it for the 'v2' protocol, though.

**Comment by bryphe** [comment]
> I wonder where the letter and color comes from now though.

It looks like they moved to the decorations API:
```
export interface DecorationRequest {
	readonly id: number;
	readonly handle: number;
	readonly uri: UriComponents;
}

export type DecorationData = [number, boolean, string, string, ThemeColor];
export type DecorationReply = { [id: number]: DecorationData; };

export interface ExtHostDecorationsShape {
	$provideDecorations(requests: DecorationRequest[], token: CancellationToken): Promise<DecorationReply>;
}

```
https://github.com/onivim/vscode-exthost/blob/50bef147f7bbd250015361a4e3cad3305f65bc27/src/vs/workbench/api/common/extHost.protocol.ts#L1502

**Comment by glennsl** [comment]
> It looks like they moved to the decorations API

I think it already was in the decorations AP. We use that for the file explorer. But yeah they probably moved to use that for the resources as well.

**Pull request merged**